### PR TITLE
Example terraform module

### PR DESCRIPTION
**Pull request by dgniewek**
Example terraform module using `null-resource`, with:
* examples
* terraform-docks
* updated README
* `.gitignore`

### CODE REVIEW

File: example/full-example/versions.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 0.13.0"
+
+  required_providers {
+    null = {
+      source  = "hashicorp/null"
+      version = "3.1.1"
```
Comment (jakubigla) [inline example/full-example/versions.tf:7]: Should we allow for patches?

**Review by jakubigla — COMMENTED** [review]
What about the naming convention for the module names and other bits? 
Shouldn't we stick everywhere to the underscore?  Happy to be convinced to the hyphens.

File: example/full-example/versions.tf (hunk 1 above)
Comment (dgniewek) [reply example/full-example/versions.tf:7]: Regarding the convention - I usually use `-` hyphens to resource names and `_` underscores for variables, as it makes sense to differentiate

Comment (dgniewek) [reply example/full-example/versions.tf:7]: *as resource names I also understand local module names :)

**Review by jakubigla — APPROVED** [review]

**Pull request merged**